### PR TITLE
block points add mode in 3D when layer.ndim == 2

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
This PR is a temporary patch for #4096 . @jni discovered that adding points in 3D to a layer with ndim == 2 is strange. As described in this [comment](https://github.com/napari/napari/issues/4096#issuecomment-1056751468), making the layers able to add new data that changes ndim is a bigger issue. For related reasons, the other 3D interactivity is strange when `layer.ndim == 2` (e.g., the selection bounding box). Thus, I propose we disable interactivity (i.e., set layer.editable to False) when `ndim ==2 and ndisplay == 3`.

I have also added an additional flag for disabling the addition mode altogether when ndim == 3. @jni correctly pointed out that the current point addition via the GUI in 3D is not intuitive because the plane the points are being added to is not depicted. It is a big hacky, but it works. I don't have a strong opinion either way if we should disable interactive point addition in 3D. However, I think it could be nice to keep in.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
This is a temporary patch for #4096 

# How has this been tested?
- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/layers/points/points.py
Comment (kevinyamauchi) [inline napari/layers/points/points.py]: This is an implementation that allows us to disable interactive point adding in 3D. If we just want to disable it when `self.ndim < 3 and self._ndisplay == 3`, it could be:

```suggestion
        if self.ndim < 3 and self._ndisplay == 3:
            self.editable = False
```

**Comment by kevinyamauchi** [comment]
I am leaning towards just disabling point addition when layer.ndim == 2 and layer._ndisplay ==3. While the current point adding in 3D is a not super intuitive, it does work and special casing it out requires more of a change. I think we can make a better point addition interface in time for the next release

It would be great to get some feedback, @jni and @alisterburt . If possible, I would like to include this in 0.1.15.

**Review by alisterburt — APPROVED** [review]
Looks great @kevinyamauchi - the ndim and the camera plane are definitely things we should tackle separately

I'm approving as is but agree that having points added on the camera plane is the way to go for now, even if unintuitive, because of the special casing required on the Qt side if we go the other way. I will leave the final decision to @jni :)

**Comment by jni** [comment]
@kevinyamauchi to clarify, I need to commit your suggestion to get the 3D addition behaviour, yes? I'm happy to go along with it!

You've missed the boat for 0.1.15 though, we are about to release 0.**4**.15! :joy:

**Review by jni — APPROVED** [review]
@alisterburt @kevinyamauchi I'm heading to bed, please feel free to merge whichever version you both agree with, both versions are a big improvement over the current behaviour.

I'll cut a new rc sometime tomorrow.

File: napari/layers/points/points.py
Comment (kevinyamauchi) [reply napari/layers/points/points.py]: Based on feedback, I have switched to just disallowing interactivity when ndim < 3 and ndisplay == 3 (i.e., point adding is allows for ndim >= 3 layers in 3D mode).

**Comment by kevinyamauchi** [comment]
Thanks for the quick feedback, @jni and @alisterburt ! Based on your feedback, we now disallow interactivity when `layer.ndim < 3 and layer._ndisplay == 3`, but allow interactivity when `layer.ndim >=3 and layer._ndisplay == 3`.  This is the most simple fix and we can improve the UI for adding points for 0.**4**.16. I also added a couple of tests.

@alisterburt , I think this is ready for merge if you are happy with the changes and the tests pass.

> You've missed the boat for 0.1.15 though, we are about to release 0.4.15! 😂

Haha, wow, I was really confused about where we are version-wise. At least I was consistent 🤷

**Review by brisvag — APPROVED** [review]
I agree, better wonky behaviour that can be used rather than none at all! And I agree on disabling the 2d/3d case, it was *too* unexpected and got me every time. LGTM!

**Comment by kevinyamauchi** [comment]
Merging now since we have 3 approvals. Thanks, everyone!

**Pull request merged**
